### PR TITLE
[Fixtures] Updated CommonBuildSettings with Xcode 8.2.1

### PR DESCRIPTION
**Pull request by louisdh**
Updated Common Build Settings with Xcode 8.2.1.
Also updated gems to fix `bundler install`.

This updates the default build settings. Including the default optimization level for Swift release targets, which is now whole module optimization.

### CODE REVIEW

**Comment by louisdh** [comment]
Spec failures now at 19 (down from 46)

**Comment by louisdh** [comment]
All spec failures have been resolved, but now Rubocop is complaining 😕

**Comment by louisdh** [comment]
CI is happy now 😪
@segiddins Ready to merge?

File: lib/xcodeproj/constants.rb [hunk 1]
```diff
@@ -192,11 +192,21 @@ module Constants
...
+      }.freeze,
       [:debug, :application, :swift] => {
         'SWIFT_OPTIMIZATION_LEVEL'          => '-Onone',
       }.freeze,
       [:debug, :swift] => {
-        'SWIFT_OPTIMIZATION_LEVEL'          => '-Onone',
+        'SWIFT_OPTIMIZATION_LEVEL'            => '-Onone',
+        'SWIFT_ACTIVE_COMPILATION_CONDITIONS' => 'DEBUG',
+      }.freeze,
+      [:release, :application, :swift] => {
```
Comment (segiddins) [inline lib/xcodeproj/constants.rb]: The separate entry for applications shouldn't be necessary

File: spec/fixtures/CommonBuildSettings/Project/Objc_OSX_Native/ViewController.m [hunk 2]
```diff
@@ -3,15 +3,17 @@
 @implementation ViewController
 
 - (void)viewDidLoad {
-    [super viewDidLoad];
+	[super viewDidLoad];
```
Comment (segiddins) [inline spec/fixtures/CommonBuildSettings/Project/Objc_OSX_Native/ViewController.m]: What's with all these indentation changes?

Comment (louisdh) [inline spec/fixtures/CommonBuildSettings/Project/Objc_OSX_Native/ViewController.m]: Ah, my Xcode settings seem to use tabs, but that doesn't really matter here does it? (basically unused code)

File: lib/xcodeproj/constants.rb (hunk 1 above)
Comment (louisdh) [inline lib/xcodeproj/constants.rb]: You're right, removed it

File: spec/fixtures/CommonBuildSettings/Project/Objc_OSX_Native/ViewController.m (hunk 2 above)
Comment (louisdh) [inline spec/fixtures/CommonBuildSettings/Project/Objc_OSX_Native/ViewController.m]: Do you want me to change the indentation to spaces?

Comment (louisdh) [inline spec/fixtures/CommonBuildSettings/Project/Objc_OSX_Native/ViewController.m]: Indentation has been changed to 4 spaces.

**Comment by louisdh** [comment]
@segiddins Anything else you'd like to see different?

**Comment by louisdh** [comment]
@segiddins Can this be merged?

**Comment by segiddins** [comment]
Looks good to me

**Comment by benasher44** [comment]
@louisdh thanks!

**Comment by benasher44** [comment]
@louisdh with the release of Xcode 8.3, we'd certainly appreciate another PR if you find that there are more settings that need updating! :)

**Comment by louisdh** [comment]
Ok, I'll take a look. I'm glad this got merged since the diff. between 8.2.1 and 8.3 will be much smaller than this PR 🙂

**Comment by benasher44** [comment]
I agree :). Thanks again for having a look!

**Pull request merged**